pinot-core/src/main/java/org/apache/pinot/core/data/table/TableResizer.java

```diff
@@ -69,40 +72,57 @@
     _orderByValueExtractors = new OrderByValueExtractor[_numOrderBy];
     Comparator[] comparators = new Comparator[_numOrderBy];
 
-    for (int orderByIdx = 0; orderByIdx < _numOrderBy; orderByIdx++) {
-      SelectionSort selectionSort = orderBy.get(orderByIdx);
-      String column = selectionSort.getColumn();
-
-      if (columnIndexMap.containsKey(column)) {
-        int index = columnIndexMap.get(column);
-        if (index < numKeyColumns) {
-          _orderByValueExtractors[orderByIdx] = new KeyColumnExtractor(index);
+    if (numKeyColumns < numColumns) {
```

Comment:
Would it be better to separate out the TableResizer for Aggr and Distinct (share as much code as possible, but separate out the custom code into different classes)?

Reply:
I wanted to think about how to extract and share common pieces from TableResizer -- since that change will result in cascading changes to IndexedTable (the one used in GROUP BY), I thought I can take a stab at it in a follow-up. May be have a hierarchy here as well just like we have for Table -> BaseTable -> multiple implementations